agent/engine/dependencygraph/graph.go

```diff
@@ -205,25 +209,25 @@ func verifyStatusResolvable(target *apicontainer.Container, existingContainers m
 // (map from name to container). The `resolves` function passed should return true if the named container is resolved.
 
 func verifyContainerOrderingStatusResolvable(target *apicontainer.Container, existingContainers map[string]*apicontainer.Container,
-	resolves func(*apicontainer.Container, *apicontainer.Container, string) bool) error {
+	resolves func(*apicontainer.Container, *apicontainer.Container, string) bool) (*apicontainer.DependsOn, error) {
 
 	targetGoal := target.GetDesiredStatus()
 	if targetGoal != target.GetSteadyStateStatus() && targetGoal != apicontainerstatus.ContainerCreated {
 		// A container can always stop, die, or reach whatever other state it
 		// wants regardless of what dependencies it has
-		return nil
+		return nil, nil
 	}
 
 	for _, dependency := range target.DependsOn {
 		dependencyContainer, ok := existingContainers[dependency.Container]
 		if !ok {
-			return fmt.Errorf("dependency graph: container ordering dependency [%v] for target [%v] does not exist.", dependencyContainer, target)
+			return nil,fmt.Errorf("dependency graph: container ordering dependency [%v] for target [%v] does not exist.", dependencyContainer, target)
```

Comment:
Not sure but do you want to return the &dependency in here?

Reply:
I don't think so, since this function is only validating that its possible to resolve.